scipy/stats/_continuous_distns.py

```diff
@@ -6505,42 +6510,47 @@ def scale_mle(loc, data):
             # and Peacock (2000), Page 175
             return (np.sum((data - loc) ** 2) / (2 * len(data))) ** .5
 
+        def loc_mle(loc, data):
+            # This implicit equation for `loc` is used when
+            # both `loc` and `scale` are free.
+            xm = data - loc
+            s1 = xm.sum()
+            s2 = (xm**2).sum()
+            s3 = (1/xm).sum()
+            return s1 - s2/(2*len(data))*s3
+
+        def loc_mle_scale_fixed(loc, scale, data):
+            # This implicit equation for `loc` is used when
+            # `scale` is fixed but `loc` is not.
+            xm = data - loc
+            return xm.sum() - scale**2 * (1/xm).sum()
+
         if floc is not None:
             # `loc` is fixed, analytically determine `scale`.
             if np.any(data - floc <= 0):
                 raise FitDataError("rayleigh", lower=1, upper=np.inf)
             else:
                 return floc, scale_mle(floc, data)
-        if fscale is not None:
-            # `scale` is fixed, but we cannot analytically determine `loc`, so
-            # we use the superclass fit method.
-            return super(rayleigh_gen,
-                         self).fit(data, *args, **kwds)
-        else:
-            # `floc` and `fscale` are not fixed. Use the analytical MLE for
-            # `scale` as a function of `loc` in numerical optimization of
-            # `loc`, injecting corresponding analytical optimum for `scale`.
 
-            # account for user provided guesses
-            loc, scale = self._fitstart(data)  # rv_continuous provided guesses
-            loc = kwds.pop('loc', loc)  # only `loc` user guesses are relevant
+        # Account for user provided guess of `loc`.
+        loc0 = kwds.get('loc')
```

Comment:
previously this was popped, now it's `get`. Any implications?

Reply:
The previous code was
```
            loc, scale = self._fitstart(data)  # rv_continuous provided guesses
            loc = kwds.pop('loc', loc)  # only `loc` user guesses are relevant
```
`self._fitstart(data)` was *always* called, but only `loc` was used, and if the user also provided `loc`, the value computed by `_fitstart` was ignored, so there was no need to have called it.  The new code only calls `_fitstart` if we know we're going to use the value of `loc` that it computes.

Reply:
... and in the updated code, `kwds` is not used after this, so using `get` is fine.